client/signup/steps/domains/index.jsx

```diff
@@ -632,6 +668,10 @@ class DomainsStep extends Component {
 		}
 
 		if ( isReskinned ) {
+			if ( this.isExperiment ) {
```

Comment:
```suggestion
			if ( ! stepSectionName && this.isExperiment ) {
```
This should fix the issue reported above about the subheader showing up in the transfer/map section